### PR TITLE
chore: switch reporters from util.inherits to ES2015 classes

### LINKED ISSUE #5025: 🐛 Bug: remove util.inherits / convert to classes
**Is your feature request related to a problem or a nice-to-have?? Please describe.**
It would be nice to get rid of some NodeJS specific things that makes it better suited for browsers.
one such thing could be the option to remove `util.inherits` and replacing it with classes that can extends things.
resulting in fewer dependencies and smaller code bundle that won't require the browserified version of [util](https://npmgraph.js.org/?q=util#select=exact%3Autil%400.12.5) that has an additional of 20 modules and 12 kb

NodeJS docs have marked it as legacy and has this to say about `util.inherits`

> Usage of util.inherits() is discouraged. Please use the ES6 class and extends keywords to get language level inheritance support

**Describe the solution you'd like**
Change prototype based classes into es-classes

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #5025
- [x] That issue was marked as [`status: accepting prs`](https://github.com/mochajs/mocha/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/mochajs/mocha/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Switches `Base`, the reporters that extend from it, and class-likes in those files.

This one is more involved than the others, as existing test helpers need to be updated. Not pushed to this PR branch yet.

### CODE REVIEW

**Comment by JoshuaKGoldberg** [comment]
Closing for now since https://github.com/mochajs/mocha/issues/5025 is blocked on us being ready for a next major version. We can reopen later.

**Comment by JoshuaKGoldberg** [comment]
Reopening now that we're working on v12 (#5357).

**Comment by JoshuaKGoldberg** [comment]
Ah, I remember this now: the unit tests needed to be massaged quite a bit too. I'll do that.

File: test/reporters/doc.spec.js [hunk 1]
```diff
@@ -32,15 +32,15 @@ describe("Doc reporter", function () {
...
+        it("should log html with indents and expected title", async function () {
           runner = createMockRunner(
             "suite",
             EVENT_SUITE_BEGIN,
             null,
             null,
             suite,
           );
-          var stdout = runReporter(this, runner, options);
+          var { stdout } = await runReporter({}, runner, options);
```
Comment (mark-wiemer) [inline test/reporters/doc.spec.js:43]: await has no effect here, nor anywhere else in the file, am I missing something?

File: test/reporters/xunit.spec.js [hunk 2]
```diff
@@ -594,69 +623,4 @@ describe("XUnit reporter", function () {
       expect(lines[0], "to contain", defaultSuiteName);
     });
   });
-
-  describe("showRelativePaths reporter option", function () {
```
Comment (mark-wiemer) [inline test/reporters/xunit.spec.js:598]: why remove this suite?

**Review by mark-wiemer — COMMENTED** [review]
Generally pretty good, just a couple points of confusion :)

**Comment by mark-wiemer** [comment]
Test failure is transient git issue, tests pass locally, all other tests pass, merging :)

File: test/reporters/doc.spec.js (hunk 1 above)
Comment (mark-wiemer) [reply test/reporters/doc.spec.js:43]: we can discuss post-v12, not critical

File: test/reporters/xunit.spec.js (hunk 2 above)
Comment (mark-wiemer) [reply test/reporters/xunit.spec.js:598]: restored this one

**Pull request closed without merging**